src/components/ScenarioParameters/components/ScenarioParametersInputs/GenericDateInput.js

```diff
@@ -21,30 +21,23 @@ export const GenericDateInput = ({ parameterData, parametersState, setParameters
     invalidDateMessage: t('genericcomponent.dateInput.error.invalidDateMessage', 'Date is invalid'),
   };
 
-  function setValue(newValue) {
-    setParametersState((currentParametersState) => ({
-      ...currentParametersState,
-      [parameterData.id]: newValue,
-    }));
-  }
-
   return (
     <BasicDateInput
       key={parameterData.id}
       id={parameterData.id}
       data-cy={`date-input-${parameterData.id}`}
       label={t(`solution.parameters.${parameterData.id}`, parameterData.id)}
       tooltipText={t(TranslationUtils.getParameterTooltipTranslationKey(parameterData.id), '')}
-      value={parametersState[parameterData.id] || new Date()}
-      changeSelectedDate={setValue}
+      value={parameterValue || new Date()}
+      changeSelectedDate={setParameterValue}
       dateProps={dateProps}
     />
   );
 };
 
 GenericDateInput.propTypes = {
   parameterData: PropTypes.object.isRequired,
-  parametersState: PropTypes.object.isRequired,
-  setParametersState: PropTypes.func.isRequired,
   context: PropTypes.object.isRequired,
+  parameterValue: PropTypes.any,
+  setParameterValue: PropTypes.func.isRequired,
```

Comment:
Does it make sense to set `setParameterValue` to _required_ but leave `parameterValue` to _non-required_ ? When `setParameterValue` will be called, there will be no feedback if `parameterValue` is not set.

Reply:
yes it make sense bacause the value can be null or undefned until the user change this.
The call of setParameterValue will set the new value and parmaterValue will be updated.
SetParameterValue can be also called with a null / undefined value.

Reply:
Ok, thank you for the explanation :slightly_smiling_face: